src/main/java/com/zombiecastlerush/building/Puzzle.java

```diff
@@ -36,11 +35,13 @@ public int getAttempts() {
 
     //Methods
     public int attemptPuzzle(String answer){
-            attempts+=1;
-            if(answer.equalsIgnoreCase(getSolution())){
-                super.setCleared(true);
-            }
-            return attempts;
+        attempts+=1;
+        if(answer.equalsIgnoreCase(getSolution())){
+            this.setCleared(true);
+        }
+        else if(getAttempts()<3)
```

Comment:
You don't seem to handle the case where `getAttempts() == 4`, but the answer is incorrect.

Reply:
i believe you have another spot that checks attempts, right? if >= 3 then terminate the attempts
if you have that, i think you can consider whether you want to check getAttempts() in here or not.

this only display message for the first 3 wrong attempts (attempt = 0, 1, 2). the 4th(attempt=3) one will be checked but won't display the wrong message.